### PR TITLE
Migration to Androidx

### PR DESCRIPTION
**Pull request by jub-jan**
Migrated to androidx

### CODE REVIEW

**Comment by rion18** [comment]
@luizgrp It seems Travis need to accept the new SDK licenses?

**Comment by jub-jan** [comment]
> @luizgrp It seems Travis need to accept the new SDK licenses?

yup, seems so. using my migrated version already in production :)

**Comment by pang0018** [comment]
@luizgrp when can we get this merged?

**Comment by Johnderate** [comment]
@jub-jan @luizgrp Any update on this? I migrated to Android X, and I depend on this library.

**Comment by matthewkrueger** [comment]
Please merge this in to the project! Your library is so convenient and has made dealing with RecyclerViews so much easier, but with androidx projects Android Studio complains about this library.

**Comment by iamayakio** [comment]
I forked and merged this project to AndroidX.
Feel free to use

 implementation 'com.github.iamayakio:SectionedRecyclerViewAdapter:1.2.1'

**Comment by luizgrp** [comment]
@jub-jan thanks for this pr!

could you please update `.travis.yml` so we can get it passing in travis?
could you also please change the base branch to `develop` instead of `master`?

File: library/build.gradle [hunk 1]
```diff
@@ -4,18 +4,18 @@ apply from: "build_files/static-code-analysis.gradle"
 ext {
     PUBLISH_GROUP_ID = 'io.github.luizgrp.sectionedrecyclerviewadapter'
     PUBLISH_ARTIFACT_ID = 'sectionedrecyclerviewadapter'
-    PUBLISH_VERSION = '1.2.0'
+    PUBLISH_VERSION = '1.2.1'
```
Comment (luizgrp) [inline library/build.gradle]: please keep it unchanged in this pr
```suggestion
    PUBLISH_VERSION = '1.2.0'
```

File: library/build.gradle [hunk 2]
```diff
@@ -4,18 +4,18 @@ apply from: "build_files/static-code-analysis.gradle"
...
+    compileSdkVersion 28
+    buildToolsVersion '28.0.3'
 
     defaultConfig {
         minSdkVersion 14
-        targetSdkVersion 27
-        versionCode 1
-        versionName "1.0"
+        targetSdkVersion 28
+        versionCode 2
```
Comment (luizgrp) [inline library/build.gradle]:
```suggestion
        versionCode 1
```

File: library/build.gradle [hunk 3]
```diff
@@ -4,18 +4,18 @@ apply from: "build_files/static-code-analysis.gradle"
...
+        versionName "1.1"
```
Comment (luizgrp) [inline library/build.gradle]:
```suggestion
        versionName "1.0"
```

File: library/build.gradle [hunk 4]
```diff
@@ -26,12 +26,12 @@ android {
...
-    compile 'com.android.support:appcompat-v7:27.1.0'
-    compile 'com.android.support:recyclerview-v7:27.1.0'
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
+    implementation 'androidx.appcompat:appcompat:1.0.0'
+    implementation 'androidx.recyclerview:recyclerview:1.0.0'
 
-    testCompile 'junit:junit:4.12'
-    testCompile "org.mockito:mockito-core:2.8.9"
+    testImplementation 'junit:junit:4.12'
+    testImplementation "org.mockito:mockito-core:2.8.9"
```
Comment (luizgrp) [inline library/build.gradle]: 👍

**Comment by luizgrp** [comment]
Addresses #134 
Closing as commit was cherry-picked to develop.

**Pull request closed without merging**